localstripe/resources.py

```diff
@@ -2359,67 +2359,71 @@ def _update(self, metadata=None, items=None, trial_end=None,
         except AssertionError:
             raise UserError(400, 'Bad request')
 
-        if items is None or len(items) != 1:
-            raise UserError(500, 'Not implemented')
+        old_plan = self.plan
+        if items is not None:
+            if len(items) != 1:
+                raise UserError(500, 'Not implemented')
 
-        # If no plan specified in update request, we stay on the current one
-        if not items[0].get('plan', None):
-            items[0]['plan'] = self.plan.id
+            # If no plan specified in update request, we stay on the current
+            # one
+            if not items[0].get('plan', None):
+                items[0]['plan'] = self.plan.id
 
-        # To return 404 if not existant:
-        Plan._api_retrieve(items[0]['plan'])
+            # To return 404 if not existant:
+            Plan._api_retrieve(items[0]['plan'])
+
+            # To return 404 if not existant:
+            if items[0]['tax_rates'] is not None:
+                [TaxRate._api_retrieve(tr) for tr in items[0]['tax_rates']]
+
+            self.quantity = items[0]['quantity']
+
+            item = SubscriptionItem(subscription=self.id,
+                                    plan=items[0]['plan'],
+                                    quantity=items[0]['quantity'],
+                                    tax_rates=items[0]['tax_rates'])
```

Comment:
Here you create a `SubscriptionItem`, but if the `if` below is `False`, it won't be used, and will stay "floating around" without being referenced in the `Subscription` object. Is there a way to avoid that? By creating it *inside* the `if`?

Reply:
You are right, done!